Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -195,9 +226,11 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner {
         compiledExec: AbsolutePath,
         writableDirectories: [AbsolutePath],
         input: PluginScriptRunnerInput,
-        textOutputHandler: @escaping (Data) -> Void,
-        handlerQueue: DispatchQueue
-    ) throws -> PluginScriptRunnerOutput {
+        observabilityScope: ObservabilityScope,
+        on queue: DispatchQueue,
+        outputHandler: @escaping (Data) -> Void,
+        completion: @escaping (Result<PluginScriptRunnerOutput, Error>) -> Void
```

Comment:
interesting that we have 2 handlers here. curious why?

Reply:
The output handler is repeatedly called as there is stdout and stderr output from the plugin process.  The fact that they are all returned as a single chunk at the end was always a temporary thing that needed to be unwound, especially for long-running plugins that call back to the plugin host.  In the next iteration that I'm preparing a PR for, this actually turns into a delegate that can receive callbacks from the plugin, including one where it emits output.  Possibly that could be part of the observability scope, but it felt specialized enough to not go through observability.

Reply:
Switched this to a delegate now.